### PR TITLE
Massive Refactor!

### PR DESCRIPTION
**Pull request by andrewvc**
Not quite ready for code review

### CODE REVIEW

**Comment by andrewvc** [comment]
Be sure to run the specs with --tag integration as well as without. This depends on setting your Gemfile to point to this PR https://github.com/logstash-plugins/logstash-mixin-rabbitmq_connection/pull/1

**Comment by andrewvc** [comment]
You'll need a gemfile like

```
source 'https://rubygems.org'
gemspec
gem "logstash-mixin-rabbitmq_connection", :path => "../logstash-mixin-rabbitmq_connection"
gem "logstash-core", :path => "/Users/andrewvc/projects/logstash"
gem "logstash-devutils", :path => "/Users/andrewvc/projects/logstash-devutils"⏎
```

File: Gemfile [hunk 1]
```diff
@@ -1,2 +1,3 @@
 source 'https://rubygems.org'
-gemspec
\ No newline at end of file
+gemspec
+gem "logstash-mixin-rabbitmq_connection", :path => "../logstash-mixin-rabbitmq_connection"
```
Comment (ph) [inline Gemfile]: We need to remove this line before publishing.

File: lib/logstash/outputs/rabbitmq.rb [hunk 2]
```diff
@@ -1,188 +1,59 @@
...
-
-  private
-  def connection_open?
-    @hare_info && @hare_info.connection && @hare_info.connection.open?
-  end
-
-  private
-  def sleep_for_retry
-    sleep @connect_retry_interval
+require "logstash/devutils/rspec/spec_helper"
```
Comment (ph) [inline lib/logstash/outputs/rabbitmq.rb]: We don't need to require the `spec_helper` in the plugin file.

Comment (andrewvc) [inline lib/logstash/outputs/rabbitmq.rb]: will remove

File: lib/logstash/outputs/rabbitmq.rb [hunk 3]
```diff
@@ -1,188 +1,59 @@
...
+require "logstash/pipeline"
+require "logstash/plugin_mixins/rabbitmq_connection"
+
+module LogStash
```
Comment (ph) [inline lib/logstash/outputs/rabbitmq.rb]: I know it doesn't say much but can you add the block of documentation before the module? The doc tool will pick it up.

```
# Push events to a RabbitMQ exchange. Requires RabbitMQ 2.x
# or later version (3.x is recommended).
#
# Relevant links:
#
# * http://www.rabbitmq.com/[RabbitMQ]
# * http://rubymarchhare.info[March Hare]

Comment (andrewvc) [inline lib/logstash/outputs/rabbitmq.rb]: Good catch!

File: README.md [hunk 4]
```diff
@@ -1,86 +1,4 @@
...
-All contributions are welcome: ideas, patches, documentation, bug reports, complaints, and even something you drew up on a napkin.
-
-Programming is not a required skill. Whatever you've seen about open source and maintainers or community members  saying "send patches or die" - you will not see that here.
-
-It is more important to the community that you are able to contribute.
-
-For more information about contributing, see the [CONTRIBUTING](https://github.com/elastic/logstash/blob/master/CONTRIBUTING.md) file.
\ No newline at end of file
+This is a mixin supporting  the logstash rabbitmq inputs and outputs. If you plan on writing a logstash plugin using
+rabbitmq this may be useful.
```
Comment (ph) [inline README.md]: I think this is the wrong readme here? :)

File: logstash-output-rabbitmq.gemspec [hunk 5]
```diff
@@ -23,7 +23,6 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "logstash-core", '>= 1.4.0', '< 2.0.0'
 
   s.platform = RUBY_PLATFORM
-  s.add_runtime_dependency 'march_hare', ['~> 2.12.0'] #(MIT license)
 
   s.add_development_dependency 'logstash-devutils'
   s.add_development_dependency 'logstash-input-generator'
```
Comment (ph) [inline logstash-output-rabbitmq.gemspec]: missing the dependency on the mixin?

File: spec/outputs/rabbitmq_spec.rb [hunk 6]
```diff
@@ -1,5 +1,4 @@
 require "logstash/devutils/rspec/spec_helper"
```
Comment (ph) [inline spec/outputs/rabbitmq_spec.rb]: Missing
```
# encoding: UTF-8

**Comment by ph** [comment]
I've made a few minor comments address them and you can merge!

excellent refactor @andrewvc !

**Pull request closed without merging**